lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -122,15 +122,15 @@ def server_version
     # @param protected_namespaces [Array<String>] Array of protected Kubernetes namespaces (defaults
     #   to KubernetesDeploy::DeployTask::PROTECTED_NAMESPACES)
     # @param render_erb [Boolean] Enable ERB rendering
-    def initialize(namespace:, context:, current_sha:, logger: nil, kubectl_instance: nil, bindings: {},
+    def initialize(namespace: nil, context:, current_sha:, logger: nil, kubectl_instance: nil, bindings: {},
       max_watch_seconds: nil, selector: nil, template_paths: [], template_dir: nil, protected_namespaces: nil,
       render_erb: true, allow_globals: false)
       template_dir = File.expand_path(template_dir) if template_dir
       template_paths = (template_paths.map { |path| File.expand_path(path) } << template_dir).compact
 
       @logger = logger || KubernetesDeploy::FormattedLogger.build(namespace, context)
       @template_sets = TemplateSets.from_dirs_and_files(paths: template_paths, logger: @logger)
-      @task_config = KubernetesDeploy::TaskConfig.new(context, namespace, @logger)
+      @task_config = KubernetesDeploy::TaskConfig.new(context, namespace, @logger, allow_globals && namespace.blank?)
```

Comment:
`allow_globals && namespace.blank?` is the result of having to support the legacy behavior of `kubernetes-deploy`.

Reply:
should we also make https://github.com/Shopify/kubernetes-deploy/pull/574/files#diff-4c33bbb177fdf882c41d892f3bde9570R144 be `allow_globals && namespace.blank?`

I'm not very sure about the `@allow_globals` and `@global_mode` in TaskConfig

Reply:
👍  Thanks for the explanation.